lib/tasks/dependencies.rake

```diff
@@ -0,0 +1,75 @@
+namespace :dependencies do 
+
+  desc 'Gather dependency data from database'
+
+  task libraries: :environment do 
+    
+    libraries = Library.select(:id, :name, :category).order(category: :asc, name: :asc)
+
+    op_types = OperationType.select(:id, :name, :category).where(:deployed => true)
+
+    users = User.select(:id, :name).all
+
+    def get_category_key(code)
+      return code.category + '/' + code.name 
+    end 
+
+    def check_libraries(protocol_content, libraries) 
+      libraries_cited = []
+      libraries.each do |library|
+        category_and_name = get_category_key(library)
+
+        if protocol_content.include?(category_and_name)
+              current_library = []
+              current_library << (category_and_name)
+              methods = get_methods(library)
+              if !methods.nil?
+                methods.each do |method|
+                  if protocol_content.include?(method) 
+                    current_library << method 
+                  end
+                end
+              end 
+            libraries_cited << current_library 
+        end
+      end 
+        return libraries_cited
+    end 
+
+    def get_methods(library)
+      library.source.content.scan(/\s{2,}def\s+([A-Za-z0-9_\.]*)/).flatten!
+    end
+    
+    def get_developers(protocol)
+      user_names = []
+      protocol.versions.each do |version|
+        if version.user_id != nil
+          user_names << User.select(:name).find(version.user_id).name 
+        end 
+      end
+      return user_names.uniq! 
+    end 
+
+    f = File.new("lib/dependencies.csv", "w") 
+    
+    f.puts "operation type category/name, [libraries cited, methods cited], number of libraries cited, #libraries within category, #libraries outside of category, developers who worked on code"
```

Comment:
Commas in a csv file are reserved for delimeters.

Reply:
I think the tabular summary data can go to CSV and the relational data should go to a JSON file.

Reply:
Maybe, but it's probably destined for a denormalized pandas dataframe anyway.